JS/Lesson35/onur/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -57,154 +62,204 @@ export const NewRecipe = () => {
 					processedData[`strMeasure${i}`] = "";
 				}
 			}
-
+			return null;
 		});
 		delete processedData.measures
 
-
 		processedData.dateModified = new Date().toString();
 
-		reset();
+		const allId = recipeContext.map(recipe => recipe.idMeal)
+		const newId = Math.max(...allId) + 1
 
-		console.log(processedData);
+		processedData.idMeal = newId
 
-	}
+		dispatch({ type: RecipeActionType.ADD_RECIPES, payload: [processedData] })
+
+
+		reset();
 
+		console.log('New recipe: ', processedData);
```

Comment:
Try to always clean console.log statements after you're done with development. In real production code it's considered a bad practice to leave console.log statements :)